### DISCUSSION TITLE
XML to KDL transform.

### DISCUSSION THREAD

**Discussion by Devasta** [kdl-org/kdl#195] [Show & Tell]
Hello all,

https://github.com/Devasta/XML2KDL

I've written an XSL transform to help folks convert their files to KDL.

https://xsltfiddle.liberty-development.net/gVAkJ52/6

Here it is transforming the kdl homepage (slightly modified to be valid XML) to KDL.

Hope someone finds it useful. I have been feeding the resulting KDL outputs into kdl4j and it has not complained so far, but if you find any issues please let me know! :)

Regards

Dan

**Comment by zkat** [comment]
This is so cool! You should send a PR here and to the website to add it to the list of implementations because I think this definitely counts as one?

**Reply by Devasta** [reply]
Thank you so much!

https://github.com/kdl-org/kdl/pull/201
https://github.com/kdl-org/kdl-org.github.io/pull/18

I have never used Github (or Git!) before really, so let me know if I need to redo these PRs. :)

**Comment by hkolbeck** [comment]
I'd love to hear any feedback you have on kdl4j 🙂

**Reply by Devasta** [reply]
So I am not a Java dev, I just happen to use IntelliJ for my XSLT work, so this was my first dipping of toes into the water but I have found the library very easy to use. The main help has been the error messages which have been great, I definitely wouldn't have been able to figure it out without them.

**Comment by bkane-msft** [comment]
https://xsltfiddle.liberty-development.net/gVAkJ52/6 is broken now, but https://xsltfiddle.liberty-development.net/bET2rY5 (from the project [homepage](https://github.com/Devasta/XML2KDL) does work. I've also made https://github.com/kdl-org/kdl/pull/301 so folks can find the link in the XML in KDL page
